src/server/req_jobobit.c

```diff
@@ -2251,6 +2251,25 @@ job_obit(struct resc_used_update *pruu, int stream)
 
 		check_block(pjob, "");		/* if block set, send word */
 
+		if ((pjob->ji_qs.ji_svrflags & JOB_SVFLG_CHKPT) &&
+			((pjob->ji_qs.ji_svrflags & JOB_SVFLG_SubJob) == 0)) {
+
+			/* non-migratable checkpoint, leave there */
+			/* and just requeue the job		      */
+
+			rel_resc(pjob);
+			ack_obit(stream, pjob->ji_qs.ji_jobid);
+			pjob->ji_qs.ji_svrflags |= JOB_SVFLG_HASRUN;
+			svr_evaljobstate(pjob, &newstate, &newsubst, 1);
+			(void)svr_setjobstate(pjob, newstate, newsubst);
+			if (pjob->ji_mom_prot == PROT_TCP)
+				svr_disconnect(pjob->ji_momhandle);
+
+			pjob->ji_momhandle = -1;
+			pjob->ji_mom_prot = PROT_INVALID;
+			return;
```

Comment:
Why do we return here?  Wouldn't we want to continue in the code and set a work task?

Reply:
No we don't need it. The job needs to be requeued.

Reply:
Okay, thanks for explaining.